src/Types.jl

```diff
@@ -1077,6 +1079,24 @@ function remove_registries(io::IO, regs::Vector{RegistrySpec})
     return nothing
 end
 
+get_status_code(response::Downloads.Response) = response.status
+function get_status_code(url::AbstractString, timeout_seconds::Real)
```

Comment:
This function raises many warning flags to me. The `istaskdone` in a loop with `sleep` and the `throwto` to a task and an empty `catch` section. I'm not super well versed in the task system but `throwto` I've only heard bad stuff about. Why would the request take so long to return?

Reply:
Yeah I definitely would like a better way to do this.

If the registry hash is not known to the Pkg server, the request never returns - it seems to hang indefinitely. So we need some kind of timeout.

E.g. try out the URLs in my comment here: https://github.com/JuliaLang/Pkg.jl/issues/2215#issuecomment-731721470

The hashes that are known to the Pkg server return immediately with 200. But the hashes that the Pkg server does not know about will hang indefinitely.

Reply:
I opened a related feature request here: https://github.com/JuliaLang/Downloads.jl/issues/70